scandium-core/src/main/java/org/eclipse/californium/scandium/DTLSConnector.java

```diff
@@ -419,40 +462,78 @@ private void receiveNextDatagramFromNetwork() throws IOException {
 		LOGGER.log(Level.FINER, "Received {0} DTLS records using a {1} byte datagram buffer",
 				new Object[]{records.size(), inboundDatagramBufferSize});
 
-		for (Record record : records) {
+		for (final Record record : records) {
 			try {
-				LOGGER.log(Level.FINEST, "Received DTLS record of type [{0}]", record.getType());
 
 				switch(record.getType()) {
+				case HANDSHAKE:
 				case APPLICATION_DATA:
-					processApplicationDataRecord(record);
-					break;
 				case ALERT:
-					processAlertRecord(record);
-					break;
 				case CHANGE_CIPHER_SPEC:
-					processChangeCipherSpecRecord(record);
-					break;
-				case HANDSHAKE:
-					processHandshakeRecord(record);
+					executor.execute(new StripedRunnable() {
+
+						@Override
+						public Object getStripe() {
+							return record.getPeerAddress();
```

Comment:
> Assuming an "attack" (simultaneous messages from different addresses for the same DTLS session), the handshake may be executed in parallel from different executors.

Hmm, not sure if this is actually an *attack* or standard functionality of DTLS, i.e. if the client can pass the VERIFY_REQUEST *and* knows the session ID *and* knows the master secret then I guess the clients are all entitled to resume the session.